torchqdynamics/mesolve/mesolve.py

```diff
@@ -96,22 +102,22 @@ def mesolve(
             'Argument `jump_ops` must be a non-empty list of tensors. Otherwise,'
             ' consider using `sesolve`.'
         )
-    jump_ops = to_tensor(jump_ops)
+    jump_ops = to_tensor(jump_ops, dtype=dtype, device=device, is_complex=True)
     jump_ops = jump_ops[None, ...] if jump_ops.ndim == 2 else jump_ops
 
     # convert rho0 to a tensor and density matrix and batch by default
-    rho0 = to_tensor(rho0)
+    rho0 = to_tensor(rho0, dtype=dtype, device=device, is_complex=True)
     if is_ket(rho0):
         rho0 = ket_to_dm(rho0)
     b_H = H_batched.size(0)
     rho0_batched = rho0[None, ...] if rho0.ndim == 2 else rho0
     rho0_batched = rho0_batched[None, ...].repeat(b_H, 1, 1, 1)  # (b_H, b_rho0, n, n)
 
     # convert t_save to a tensor
-    t_save = torch.as_tensor(t_save)
+    t_save = torch.as_tensor(t_save, device=device)
```

Comment:
Also use `dtype=to_real(dtype)` here where `to_real` is imported as `from torch._prims_common import corresponding_real_dtype as to_real` so that all dtypes match.